catalyst_voices/apps/voices/lib/pages/campaign/admin_tools/campaign_admin_tools_dialog.dart

```diff
@@ -0,0 +1,309 @@
+import 'package:catalyst_voices/common/ext/space_ext.dart';
+import 'package:catalyst_voices/pages/campaign/admin_tools/campaign_admin_tools_events.dart';
+import 'package:catalyst_voices/pages/campaign/admin_tools/campaign_admin_tools_views.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// A draggable [CampaignAdminToolsDialog],
+/// should be used as a child of a [Stack].
+///
+/// Initially shown at bottom-left corner with [initialPadding] offset.
+class DraggableCampaignAdminToolsDialog extends StatefulWidget {
+  /// The key for the [CampaignAdminToolsDialog] to make sure it's state
+  /// is kept when a user keeps dragging it.
+  ///
+  /// The state might include currently open tab from [TabBar],
+  /// scroll position, etc.
+  final GlobalKey dialogKey;
+
+  /// See [CampaignAdminToolsDialog.selectedSpace].
+  final Space selectedSpace;
+
+  /// See [CampaignAdminToolsDialog.onSpaceSelected].
+  final ValueChanged<Space> onSpaceSelected;
+
+  /// See [CampaignAdminToolsDialog.onClose].
+  final VoidCallback onClose;
+
+  /// The initial offset from bottom-left for the dialog.
+  ///
+  /// The widget should be used inside a stack so left/bottom properties
+  /// from [EdgeInsets] correspond to left/bottom properties
+  /// from the related [Positioned] widget.
+  final EdgeInsets initialPadding;
+
+  const DraggableCampaignAdminToolsDialog({
+    super.key,
+    required this.dialogKey,
+    required this.selectedSpace,
+    required this.onSpaceSelected,
+    required this.onClose,
+    this.initialPadding = const EdgeInsets.only(left: 32, bottom: 32),
+  });
+
+  @override
+  State<DraggableCampaignAdminToolsDialog> createState() =>
+      _DraggableCampaignAdminToolsDialogState();
+}
+
+class _DraggableCampaignAdminToolsDialogState
+    extends State<DraggableCampaignAdminToolsDialog> {
+  late Offset _position;
+  late Size _screenSize;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    _screenSize = MediaQuery.sizeOf(context);
+    _position = Offset(
+      widget.initialPadding.left,
+      _screenSize.height -
+          CampaignAdminToolsDialog._height -
+          widget.initialPadding.bottom,
+    );
```

Comment:
Its potential bug here. When user is dragging it and `screen_size` changes `_position` will be overriden to default value from what was set in `_onDragUpdate`.

Reply:
Not sure I understood but I only see one potential bug that if you dragged your dialog to the right edge of the screen and you start to resize the browser window the dialog would teleport itself to the default position (which is bottom left corner. I have changed now the implementation to prevent it but also to make sure the dialog fits into the resized browser window